### PR TITLE
Parameterizing orka task images

### PR DESCRIPTION
**Pull request by hokadiri**
# Changes

**What:**
Parameterizing orka task images so it can be overriden

**Why:**
docker.io not accessible from internal networks. one needs to either access it via a proxy OR copy the image into an internal docker registry. Doing this requires modifying the `image` var.  

**How:**
Creating params var for task image which defaults to orka docker.io image.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by afrittoli** [comment]
@chmouel @vdemeester would this change warrant a new version of the task?
The CLA is being fixed by as a CCLA by @hokadiri's company.

**Comment by chmouel** [comment]
this doesn't break anything, so I don't think it's needed....

**Comment by afrittoli** [comment]
@tektoncd/catalog-maintainers cla fixed now, ready for review 🙇

File: task/orka-deploy/0.1/README.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ See also: [use-ssh-key.yaml](https://github.com/tektoncd/catalog/blob/main/task/
 | `verbose` | Enables verbose logging for all connection activity to the VM. | false |
 | `ssh-key` | Specifies whether the SSH credentials secret contains an [SSH key](#using-an-ssh-key), as opposed to a password. | false |
 | `delete-vm` | Specifies whether to delete the VM after use when run in a pipeline. You can discard build agents that are no longer needed to free up resources. Set to false if you intend to clean up VMs after use manually. | true |
+| `task-image` | The docker image used to run the task steps. | --- |
```
Comment (piyush-garg) [inline task/orka-deploy/0.1/README.md:94]: `---` should be replaced by default value

File: task/orka-full/0.1/README.md [hunk 2]
```diff
@@ -118,6 +118,7 @@ See also: [`use-ssh-key`](https://github.com/tektoncd/catalog/blob/main/task/ork
 | Parameter | Description | Default |
 | --- | --- | ---: |
 | `base-image` | The Orka base image to use for the VM config. | --- |
+| `task-image` | The docker image used to run the task steps. | --- |
```
Comment (piyush-garg) [inline task/orka-full/0.1/README.md:121]: `---` should be replaced by default value

File: task/orka-init/0.1/README.md [hunk 3]
```diff
@@ -135,6 +135,7 @@ Omit `--namespace` if installing in the `default` namespace.
 | Parameter | Description | Default |
 | --- | --- | ---: |
 | `base-image` | The Orka base image to use for the VM config. | --- |
+| `task-image` | The docker image used to run the task steps. | --- |
```
Comment (piyush-garg) [inline task/orka-init/0.1/README.md:138]: `---` should be replaced by default value

File: task/orka-teardown/0.1/README.md [hunk 4]
```diff
@@ -51,3 +51,4 @@ See the `orka-init` documentation [here](https://github.com/tektoncd/catalog/blo
 | `orka-token-secret-key` | The name of the key in the Orka token secret, which holds the authentication token. Applicable to `orka-init` / `orka-deploy` / `orka-teardown`. | token |
 | `orka-vm-name-config` | The name of the config map, which stores the name of the generated VM configuration. Applicable to `orka-init` / `orka-deploy` / `orka-teardown`. | orka-vm-name |
 | `orka-vm-name-config-key` | The name of the key in the VM name config map, which stores the name of the generated VM configuration. Applicable to `orka-init` / `orka-deploy` / `orka-teardown`. | vm-name |
+| `task-image` | The docker image used to run the task steps. | --- |
```
Comment (piyush-garg) [inline task/orka-teardown/0.1/README.md:54]: `---` should be replaced by default value

File: task/orka-deploy/0.1/README.md (hunk 1 above)
Comment (piyush-garg) [inline task/orka-deploy/0.1/README.md:94]: I think we should have the parameter like `orka-image` instead of `task-image`

**Review by piyush-garg — CHANGES_REQUESTED** [review]

**Comment by piyush-garg** [comment]
Please squash the commits also

**Comment by piyush-garg** [comment]
@hokadiri Please squash the commits into one

**Comment by hokadiri** [comment]
@piyush-garg ,done

**Comment by vinamra28** [comment]
@hokadiri, I guess the commits are still not squashed 😅, still 10 commits 😅

**Comment by hokadiri** [comment]
hmm, 
@vinamra28 
Maybe i'm not familiar squashing commits on github's.
I squashed via the did interactive rebase `git rebase -i HEAD~[X]`. and got this commit https://github.com/tektoncd/catalog/pull/939/commits/bce59dbd16d6ecf58d7347d01dab5f27f36b69fb.

is there a different way of squashing?

**Comment by vinamra28** [comment]
closing in favour of #952

**Pull request closed without merging**